### PR TITLE
Move hdf.object classes to their own plugin and fragment

### PR DESCRIPTION
**Pull request by belkassaby**
Signed-off-by: Baha El-Kassaby <baha.elkassaby@gmail.com>

### CODE REVIEW

File: org.eclipse.dawnsci.hdf5.test/src/org/eclipse/dawnsci/hdf5/ReadThreadTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		HDF5FileFactory.releaseFile(file);
+		now += System.nanoTime();
+
+		System.err.println("Thd for " + dataName + " took " + now*1e-9 + "s");
+	}
+
+	private void prepareForTest(String file, int nthreads) throws Exception {
+		HDF5FileFactory.deleteFile(file);
+
+		HDF5FileFactory.acquireFile(file, true);
```
Comment (PeterC-DLS) [inline org.eclipse.dawnsci.hdf5.test/src/org/eclipse/dawnsci/hdf5/ReadThreadTest.java]: Delete this line

Comment (belkassaby) [reply org.eclipse.dawnsci.hdf5.test/src/org/eclipse/dawnsci/hdf5/ReadThreadTest.java]: Done. Can I merge?

**Comment by belkassaby** [comment]
could you change your review to a green check? :)

**Comment by PeterC-DLS** [comment]
@belkassaby: which repos does this affect?

**Comment by PeterC-DLS** [comment]
Need to synchronise with these:
- dawn-commandserver.git
- dawn-common.git
- dawnsci.git
- dawn-ui.git
- dawn-workflow.git
- gda-dls-beamlines-xas.git
- gda-epics.git
- gda-tomography.git
- scisoft-arpes.git
- scisoft-core.git
- scisoft-ncd.git
- scisoft-spectroscopy.git

**Comment by belkassaby** [comment]
Hi Pete, 
It's been done for 
* dawn-common: https://github.com/DawnScience/dawn-common/pull/16
* dawn-ui: https://github.com/DawnScience/dawn-ui/pull/83
* dawnsci: https://github.com/eclipse/dawnsci/pull/598 (this pull request)
* dawn-workflow: https://github.com/DawnScience/dawn-workflow/pull/9
For this change, I don't think there needs to be any manifest changes in scisoft-ui and core (only for the IOpenFileAction change, which is something different).
You need to do it for 
* gda-dls-beamlines-xas.git
* gda-epics.git
* gda-tomography.git
* scisoft-arpes.git
* scisoft-ncd.git
* scisoft-spectroscopy.git
as I don't have access to them.

**Comment by PeterC-DLS** [comment]
Closing this for #604

**Pull request closed without merging**